### PR TITLE
Dev front

### PR DESCRIPTION
**Pull request by Ronan-UP**
Moving the new front end to dev.

### CODE REVIEW

File: SmartContract/migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const Verifier = artifacts.require("Verifier");
 const UnisonToken = artifacts.require("UnisonToken");
 const RandomSource = artifacts.require("Utilities/RandomSource.sol");
+const Faucet = artifacts.require("Faucet.sol")
```
Comment (bitBadger8) [inline SmartContract/migrations/2_deploy_contracts.js:4]: @Kaleb-Bruwer are these changes fine?

File: SmartContract/src/Faucet.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
```
Comment (Cobus-Redelinghuys) [inline SmartContract/src/Faucet.sol:1]: Why are these here?

Comment (Cobus-Redelinghuys) [reply SmartContract/src/Faucet.sol:1]: These files

File: SmartContract/test/Verifier.test.js [hunk 3]
```diff
@@ -44,30 +44,68 @@ contract('Verifier', (accounts) =>{
...
+            needCoins = [];
+            for(var i = 1; i<9; i++){
+                needCoins.push(accounts[i]);
+            }
+            giveJurorsCoins(token, accounts[0], needCoins, 100000);
+        })
+
+        it("Can't create agreement without payment", async () =>{
+
+            // try{
```
Comment (Cobus-Redelinghuys) [inline SmartContract/test/Verifier.test.js:64]: Maybe remove these comments

File: SmartContract/migrations/2_deploy_contracts.js (hunk 1 above)
Comment (Kaleb-Bruwer) [reply SmartContract/migrations/2_deploy_contracts.js:4]: These are my changes, which are already on dev. I don't know why git marked it as a change

File: frontend_scv/assets/Todo.dart [hunk 4]
```diff
@@ -1,16 +1,13 @@
...
-* Generic API Handler (Hairdu example)
-* Add payment condition to blockchain (Based on backend payment condition)
-* Touch Juror screen's Action Area
-* Pay Platform once accepted on blockchain
-* Duration input -> DatePicker
-* Jury chat --> Double Check
-* In the contacts screen, account for uppercase to lowercase walletID conversion
-* Also restyle contacts screen.
-* TODO: Go through all TODO's
+* Evidence - touch up, make Judges not able to add evidence
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/assets/Todo.dart:3]: Is this list not suppose to be empty?

File: frontend_scv/lib/models/blockchainAgreement.dart [hunk 5]
```diff
@@ -57,30 +76,17 @@ class BlockchainAgreement {
...
-//           break;
-//   case 7: ret = 'Contested';
-//           break;
-//   case 8: ret = 'Decided';
-//           break;
-//   case 9: ret = 'Closed';
-//           break;
-//
-// }
+  ///Is the current user the one who should pay? This does not account for the payment already being made. Use shouldPay to check if payment has been made
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/models/blockchainAgreement.dart:83]: has this been implemented?

File: frontend_scv/assets/Todo.dart [hunk 6]
```diff
@@ -1,16 +1,13 @@
...
+* Remove prints to console
+* Login
+* Go through all TODO - mainly in Services
+* User Feedback
+*
+* More button looking actions on action area
+* SideBar (later)
+* Agreement Status/Notification (later)
```
Comment (Kaleb-Bruwer) [inline frontend_scv/assets/Todo.dart:11]: Should this be empty?

File: frontend_scv/lib/screens/auth_screen.dart [hunk 7]
```diff
@@ -147,12 +148,19 @@ class _AuthCardState extends State<AuthCard> {
   }
 
   Future<void> _submit2() async {
-    JudgeService judgeService = JudgeService();
-    UnisonService unisonService = UnisonService();
-    await await Provider.of<Auth>(context, listen: false).metaMaskLogin();
-    await unisonService.getAgreement(BigInt.from(0));
-    await judgeService.isJudge();
-    //await judgeService.setContractAllowance();
+    // JudgeService judgeService = JudgeService();
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/screens/auth_screen.dart:151]: Should this commented-out code not be removed?

File: frontend_scv/lib/models/contract.dart [hunk 8]
```diff
@@ -1,29 +1,30 @@
...
   DateTime sealedDate;
   bool movedToBlockchain;
-  List<Condition> conditions; //TODO Handle empty/initial conditions
+  List<Condition> conditions;
   String title;
 
   String description;
   String payingUser;
-  double price;
+  double paymentAmount; //In UNT
```
Comment (Kaleb-Bruwer) [inline frontend_scv/lib/models/contract.dart:22]: This causes a precision loss, we will need to address it in the future

File: frontend_scv/lib/screens/view_assignment_screen.dart [hunk 9]
```diff
@@ -22,31 +18,19 @@ class ViewAssignmentScreen extends StatefulWidget {
...
-    //final contractId = ModalRoute.of(context).settings.arguments as String;
-    final testCon = ModalRoute.of(context).settings.arguments as Contract;
-    final loadedContract = testCon;
-
+    final loadedContract =
+        ModalRoute.of(context).settings.arguments as Contract;
     return Scaffold(
       appBar: AppBar(
-        title: Text(loadedContract.title),
+        title: FunkyText('Current Assignment'),
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/screens/view_assignment_screen.dart:33]: Why is it called funkyText

File: frontend_scv/assets/Todo.dart (hunk 4 above)
Comment (bitBadger8) [reply frontend_scv/assets/Todo.dart:3]: Yes, but we didn't have 'empty Todo list' on the todo list

File: frontend_scv/lib/services/Server/negotiationService.dart [hunk 10]
```diff
@@ -54,70 +50,50 @@ class NegotiationService {
...
-  //     print(e);
-  //     throw e;
-  //   }
-  // }
-
 
   //The api will be updated with a new 'seal flow'
   Future<void> sealAgreement(Contract con) async {
     await _uniServ.saveAgreement(con);
+    //TODO: Add payment conditions here? After the initial save, it should wait until completion.
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/services/Server/negotiationService.dart:96]: Has this todo been solved or implemented?

File: frontend_scv/lib/services/Server/negotiationService.dart [hunk 11]
```diff
@@ -54,70 +50,50 @@ class NegotiationService {
...
+    //RFC: Maybe "payment condition added" should be a check on agreement load.
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/services/Server/negotiationService.dart:97]: The same goes for this line?

File: frontend_scv/lib/screens/view_assignment_screen.dart (hunk 9 above)
Comment (bitBadger8) [reply frontend_scv/lib/screens/view_assignment_screen.dart:33]: GOT FUNK?
image

File: frontend_scv/lib/widgets/agreement/contract_action_area.dart [hunk 12]
```diff
@@ -59,17 +59,28 @@ class _ContractActionAreaState extends State<ContractActionArea> {
...
         loaderType: AwesomeLoader.AwesomeLoader4,
         color: Colors.pink,
       );
     }
 
-    if (widget._agreement.movedToBlockchain == false ||
+    if (widget._agreement.movedToBlockchain ==
+            false || //TODO: Ronan thinks Kevin should change this (Maybe to a null check?)
         widget._agreement.blockchainId == BigInt.from(-1)) {
+      //TODO: The second condition will always fail, an exception will be thrown since Bigints are always positive
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/widgets/agreement/contract_action_area.dart:74]: Has the todo been solved or implemented?

File: frontend_scv/lib/widgets/agreement/contract_action_area.dart [hunk 13]
```diff
@@ -59,17 +59,28 @@ class _ContractActionAreaState extends State<ContractActionArea> {
...
     if (_isLoading) {
       return AwesomeLoader(
         loaderType: AwesomeLoader.AwesomeLoader4,
         color: Colors.pink,
       );
     }
 
-    if (widget._agreement.movedToBlockchain == false ||
+    if (widget._agreement.movedToBlockchain ==
+            false || //TODO: Ronan thinks Kevin should change this (Maybe to a null check?)
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/widgets/agreement/contract_action_area.dart:72]: Has the todo been solved or implemented?

File: frontend_scv/lib/widgets/agreement/upload_evidence_panel.dart [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+  GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  Future<void> uploadEvidence() async {
+    FilePickerResult picked =
+        await FilePicker.platform.pickFiles(type: FileType.image);
+
+    List<PlatformFile> fileRes = picked.files;
+    Evidence evid = Evidence(widget.agreementId);
+    await evid.setFile(
+        fileRes[0]); //May be expanded in the future to allow multiple files
```
Comment (Cobus-Redelinghuys) [inline frontend_scv/lib/widgets/agreement/upload_evidence_panel.dart:27]: Is this for project day?

**Review by Cobus-Redelinghuys — APPROVED** [review]
Please just note the comments I have left
Looks good to me

File: frontend_scv/lib/widgets/contact/ContactList_item.dart [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+class ContactListItem extends StatelessWidget {
+  final ContactList list;
+  ContactListItem(this.list);
+
+  @override
+  Widget build(BuildContext context) {
+    Widget contactListItem = GestureDetector(
+      onTap: () {
+        print('A tap');
```
Comment (Kaleb-Bruwer) [inline frontend_scv/lib/widgets/contact/ContactList_item.dart:16]: ???

**Review by bitBadger8 — APPROVED** [review]
Here we go gents


image

File: frontend_scv/lib/models/blockchainAgreement.dart (hunk 5 above)
Comment (bitBadger8) [reply frontend_scv/lib/models/blockchainAgreement.dart:83]: Yessir - when making a payment condition, the user selects who pays

**Review by Kaleb-Bruwer — APPROVED** [review]

File: frontend_scv/lib/services/Server/negotiationService.dart (hunk 11 above)
Comment (Ronan-UP) [reply frontend_scv/lib/services/Server/negotiationService.dart:97]: Yeah, fixed.

File: frontend_scv/lib/widgets/agreement/contract_action_area.dart (hunk 12 above)
Comment (Ronan-UP) [reply frontend_scv/lib/widgets/agreement/contract_action_area.dart:74]: No.

File: frontend_scv/lib/widgets/agreement/contract_action_area.dart (hunk 13 above)
Comment (Ronan-UP) [reply frontend_scv/lib/widgets/agreement/contract_action_area.dart:72]: No.

File: frontend_scv/lib/widgets/agreement/upload_evidence_panel.dart (hunk 14 above)
Comment (Ronan-UP) [reply frontend_scv/lib/widgets/agreement/upload_evidence_panel.dart:27]: Maybe, probably not.

File: frontend_scv/assets/Todo.dart (hunk 6 above)
Comment (Ronan-UP) [reply frontend_scv/assets/Todo.dart:11]: Final polish for project day.

File: frontend_scv/lib/models/contract.dart (hunk 8 above)
Comment (Kaleb-Bruwer) [reply frontend_scv/lib/models/contract.dart:22]: It's fine if it's only for display purposes, though

Comment (Ronan-UP) [reply frontend_scv/lib/models/contract.dart:22]: This allows expressing the amount of currency as a UNT unit, instead of gwei.

File: frontend_scv/lib/widgets/contact/ContactList_item.dart (hunk 15 above)
Comment (Ronan-UP) [reply frontend_scv/lib/widgets/contact/ContactList_item.dart:16]: An unnecessary print, should be removed in the future.
